msal/src/test/java/com/microsoft/identity/client/robolectric/tests/AcquireTokenMockTest.java

```diff
@@ -0,0 +1,577 @@
+            void makeAcquireTokenCall(final IPublicClientApplication publicClientApplication,
+                                      final Activity activity) {
+
+                String username = "fake@test.com";
+
+                final AcquireTokenSilentParameters silentParameters = new AcquireTokenSilentParameters.Builder()
+                        .withScopes(Arrays.asList(SCOPES))
+                        .forceRefresh(false)
+                        .fromAuthority(AAD_MOCK_AUTHORITY)
+                        .callback(new AuthenticationCallback() {
+                            @Override
+                            public void onSuccess(IAuthenticationResult authenticationResult) {
+                                Assert.assertTrue(!StringUtil.isEmpty(authenticationResult.getAccessToken()));
+                            }
+
+                            @Override
+                            public void onError(MsalException exception) {
+                                fail(exception.getMessage());
+                            }
+
+                            @Override
+                            public void onCancel() {
+                                fail("User cancelled flow");
+                            }
+                        })
+                        .build();
+
+                AcquireTokenParameters parameters = new AcquireTokenParameters.Builder()
+                        .startAuthorizationFromActivity(activity)
+                        .withLoginHint(username)
+                        .withScopes(Arrays.asList(SCOPES))
+                        .fromAuthority(AAD_MOCK_AUTHORITY)
+                        .callback(new AuthenticationCallback() {
+                            @Override
+                            public void onSuccess(IAuthenticationResult authenticationResult) {
+                                Assert.assertTrue(!StringUtil.isEmpty(authenticationResult.getAccessToken()));
+                                IAccount account = authenticationResult.getAccount();
+                                silentParameters.setAccount(account);
+                            }
+
+                            @Override
+                            public void onError(MsalException exception) {
+                                fail(exception.getMessage());
+                            }
+
+                            @Override
+                            public void onCancel() {
+                                fail("User cancelled flow");
+                            }
+                        })
+                        .build();
+
+                publicClientApplication.acquireToken(parameters);
+                RoboTestUtils.flushScheduler();
+                publicClientApplication.acquireTokenSilentAsync(silentParameters);
+                RoboTestUtils.flushScheduler();
+            }
+
+        }.performTest();
+    }
+
+    @Test
+    public void canAcquireSilentIfValidTokensAvailableInCache() {
```

Comment:
Where are we injecting the token into the cache?

Reply:
In this test case we are simulating that a successful acquire token call was made some time ago, and since that call was successful, we should have valid token in the cache because tokens are saved to the cache at the end of a successful acquire token call. Now we try do a silent call and it should succeed because we have valid (unexpired) tokens in the cache.